internal/eventapi/serving/subscription_manager.go

```diff
@@ -0,0 +1,248 @@
+package serving
+
+import (
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/clock"
+
+	"github.com/G-Research/armada/internal/eventapi/eventdb"
+	"github.com/G-Research/armada/internal/eventapi/model"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// SubscriptionManager lets callers subscribe to channels of events in an efficient manner
+type SubscriptionManager interface {
+	Subscribe(jobset int64, fromOffset int64) *model.EventSubscription
+	Unsubscribe(subscriptionId int64)
+}
+
+type internalSubscription struct {
+	subscriptionId int64
+	jobset         int64
+	channel        chan []*model.EventRow
+}
+
+type DefaultSubscriptionManager struct {
```

Comment:
If you don't want to expose the default implementation, I suggest calling it "subscriptionManager" and creating a method "New" with return type "SubscriptionManager" (i.e., the interface type).

Reply:
So it's nothing to do with not wanting to hide the internals (all the internal stuff is module private- anyway).  The reason for the interface type is purely for test purposes so that we can easily inject a mock/dummy subscription manager that is much simpler than this.